### ISSUE TITLE
Multi task loss, combine regularization losses

### ISSUE THREAD

**Issue by gionanide** [ywatanabe1989/custom_losses_pytorch#2]
First of all thank you for the function, also I want to express some questions though I do not know if this is the right place.

My first question is if I want to combine a Classification loss, a Regression loss, and some other losses like orthogonality loss which penalize the model when the weight matrix is not orthogonal, and L2 regularization, or a Coverage loss in NLP. This losses are inputs to the function? or only the Classification and Regression losses are inputs to the function and I just add the others?

Also I want to ask how to balance the two tasks if, let's say the regression task converges much faster than the other task.

Thank you very much.

**Comment by ywatanabe1989** [comment]
Thank you for your comment.

The Multi-task loss is introduced by Kendall et al. (https://arxiv.org/abs/1705.07115).
That balances the importance (weights) of the tasks based on their uncertainties.
In my understanding, the mechanism estimates the uncertainty parameter (sigmas) for each loss fluctuation during training, just like which to and how long to change the fulcrum of a seesaw! That's why this system is efficient to optimize for several tasks at the same time. 

According to the paper, the multi-task loss is proven to be mathematically effective only for regression and classification losses, though this does not mean that it is not useful or harmful for other types of losses at all. So, if I were in your shoes, I would try the following pseudo-codes for safety.
```
multi_task_loss = MultiTaskLoss(the classification loss, the regression loss)
total_loss = multi_task_loss + orthogonality_loss (?) + L2_loss
total_loss.backward()
```
However, the mult-task loss scales calssification and regression losses
https://github.com/ywatanabe1989/custom_losses_pytorch/blob/87eddc4277e0352824b11a023cc377d61b7b903a/multi_task_loss.py#L17-L18
So, the balance between other losses must change and the parameter tuning like for L2 lambda should be re-executed from a starting point.

All the best :+1:

**Comment by gionanide** [comment]
Thank you for the reply, you answer was clarifying.
